### PR TITLE
feat(unchained-client): parse Avalanche 0x transactions

### PR DESCRIPTION
**Pull request by 0xApotheosis**
Adds the `zrx` parser to the Avalanche parser at `evm/avalanche/parser/index.ts`.

I also lifted the `zrx` parser to `evm/parser/zrx.ts` for reuse, as it's not specific to any EVM.

Avalanche trades now show correctly in `web` (previously they were generic "transactions" as we were missing `trade` data).

Screen Shot 2022-08-08 at 3 25 42 pm

Screen Shot 2022-08-08 at 3 39 58 pm

Closes https://github.com/shapeshift/lib/issues/951

### CODE REVIEW

File: packages/unchained-client/src/evm/ethereum/parser/thor.ts [hunk 1]
```diff
@@ -32,16 +32,16 @@ export class Parser implements SubParser<Tx> {
...
-      case 'eip155:1':
-        this.routerContract = THOR_ROUTER_CONTRACT_MAINNET
-        break
-      case 'eip155:3':
-        this.routerContract = THOR_ROUTER_CONTRACT_ROPSTEN
-        break
-      default:
-        throw new Error('chainId is not supported. (supported chainIds: eip155:1, eip155:3)')
-    }
+    this.routerContract = (() => {
```
Comment (0xApotheosis) [inline packages/unchained-client/src/evm/ethereum/parser/thor.ts:35]: Just a refactor to use an IIFE.

File: packages/unchained-client/src/evm/ethereum/parser/uniV2.ts [hunk 2]
```diff
@@ -41,16 +41,16 @@ export class Parser implements SubParser<Tx> {
...
-      case 'eip155:1':
-        this.wethContract = WETH_CONTRACT_MAINNET
-        break
-      case 'eip155:3':
-        this.wethContract = WETH_CONTRACT_ROPSTEN
-        break
-      default:
-        throw new Error('chainId is not supported. (supported chainIds: eip155:1, eip155:3)')
-    }
+    this.wethContract = (() => {
```
Comment (0xApotheosis) [inline packages/unchained-client/src/evm/ethereum/parser/uniV2.ts:44]: Just a refactor to use an IIFE.

File: packages/unchained-client/src/evm/ethereum/parser/weth.ts [hunk 3]
```diff
@@ -28,16 +28,16 @@ export class Parser implements SubParser<Tx> {
...
-      case 'eip155:1':
-        this.wethContract = WETH_CONTRACT_MAINNET
-        break
-      case 'eip155:3':
-        this.wethContract = WETH_CONTRACT_ROPSTEN
-        break
-      default:
-        throw new Error('chainId is not supported. (supported chainIds: eip155:1, eip155:3)')
-    }
+    this.wethContract = (() => {
```
Comment (0xApotheosis) [inline packages/unchained-client/src/evm/ethereum/parser/weth.ts:31]: Just a refactor to use an IIFE.

**Review by gomesalexandre — APPROVED** [review]
Tested locally and this brings parsing of Avalanche 0x Txs, correctly bringing over the swapper information (order route and transaction type):

main:

image


This diff:

image

Txs being parsed as trades is already fixed when using `main` unchained-client, following https://github.com/shapeshift/lib/pull/950

**Pull request merged**